com.ibm.wala.cast.js.rhino/src/main/java/com/ibm/wala/cast/js/translator/RhinoToAstTranslator.java

```diff
@@ -1225,6 +1225,13 @@ private String getParentName(AstNode fn) {
 
     @Override
     public CAstNode visitFunctionNode(FunctionNode fn, WalkContext context) {
+      // Declare a unique "this" bind to the current context's this
+      String thisName = "this_" + context.top().hashCode();
```

Comment:
I wouldn't use `hashCode()` here, as it will be non-deterministic and make debugging harder.  There should only be one of these variables declared per scope, and we don't care about being able to serialize back to JS code.  So you could use a name like `%%saved_this`, which is guaranteed not to create name conflicts since `%` is not a legal character for a JS identifier.  You could then add a counter variable at the end, like `%%saved_this_1`, etc., with the counter stored as an instance field in `RhinoToAstTranslator`.